toucan_connectors/revinate/revinate_connector.py

```diff
@@ -0,0 +1,138 @@
+    The datasource for Revinate
+
+    Must have a valid Revinate endpoint field otherwise the connector won't know which endpoint to call
+    The params field is optional but has to be a valid Revinate query field
+
+    cf. https://porter.revinate.com/documentation
+
+    Contains:
+    - an endpoint (required)
+    - optional params
+    - a JQ filter (required)
+    """
+
+    endpoint: str = Field(
+        ...,
+        description='A valid Revinate endpoint (eg. hotelsets), cf. Resources on https://porter.revinate.com/documentation',
+    )
+    params: dict = Field(
+        None,
+        description='JSON object of valid Revinate parameters to send in the query string of this HTTP request (eg. {"page": 2, "size": 20}, which generates a query like https://porter.revinate.com/hotelsets?page=2&size=20), cf. https://porter.revinate.com/documentation',
+    )
+    filter: str = FilterSchema
+
+
+class RevinateConnector(ToucanConnector):
+    """
+    The main Revinate connector
+
+    - It's async
+    - It returns a basic pandas Dataframe based on whatever JQ filter is passed to it or it returns an error
+    """
+
+    data_source_model: RevinateDataSource
+    authentication: RevinateAuthentication
+
+    baseroute = 'https://porter.revinate.com'
+
+    async def _get_data(self, query, jq_filter):
+        """
+        Basic data retrieval function
+
+        - builds the full url
+        - retrieves built headers
+        - calls a fetch and returns filtered data or an error
+        """
+        full_url = f'{self.baseroute}/{query}'
+        api_key = self.authentication.api_key
+        api_secret: str = self.authentication.api_secret.get_secret_value()
+        username = self.authentication.username
+        timestamp = int(datetime.datetime.now().timestamp())
+
+        headers = build_headers(api_key, api_secret, username, str(timestamp))
+
+        async with ClientSession(headers=headers) as session:
+            data = await fetch(full_url, session)
+
+            try:
+                return transform_with_jq(data['content'], jq_filter)
+            except ValueError:
+                # This follows the HTTP connector's behavior for a similar situation
+                LOGGER.error('Could not transform the data using %s as filter', jq_filter)
+                raise
+            except ScriptRuntimeError:
+                LOGGER.error('Could not transform the data using %s as filter', jq_filter)
+                raise
+
+    def _run_fetch(self, query, jq_filter):
+        """Event loop handler"""
+        loop = get_loop()
+        future = asyncio.ensure_future(self._get_data(query, jq_filter))
+        return loop.run_until_complete(future)
+
+    def _retrieve_data(self, data_source: RevinateDataSource) -> pd.DataFrame:
+        """
+        Primary function and point of entry
+        """
+        query = data_source.endpoint
+
+        query = nosql_apply_parameters_to_query(query=query, parameters=data_source.params)
```

Comment:
I think we should rename query into endpoint everywhere
query made me think that there was some kind of data sent into the body of the request, while endpoint is very clear of what it actually is (a part the URL)